### PR TITLE
core: disable debug log level when the server restarts. resolves #8315

### LINKED ISSUE #8315: Enabling/disabling external access or proxy, while enhanced logging is enabled, unticks enhanced logging but leaves it enabled
0.16.103.0 Dockers and Windows sandbox. Happens on earlier versions too.

Steps to reproduce:
1. enable enhanced logging, apply - enhanced logging is ticked on refresh
2. run test - log shows enhanced logging working
3. enable or disable external access (or proxy), apply - enhanced logging is unticked on refresh
4. run test - log shows enhanced logging still enabled (running multiple tests doesn't change this)
5. apply server setting again or restart Jackett - enhanced logging is unticked on refresh
6. run test - log shows enhanced logging now disabled

Also occurs if you enable enhanced logging and external access or proxy at the same time:
1. enable enhanced logging and external access (or proxy), apply - enhanced logging is unticked on refresh
2. run test - log shows enhanced logging working
3. apply server settings again or restart Jackett - enhanced logging is unticked on refresh
4. run test - log shows enhanced logging now disabled

Not sure if the intended behaviour is to disable enhanced logging on these changes, and that's not happening, or if enhanced logging is supposed to remain enabled and it's simply a webUI error.

Disable auto update and Update to pre-release do not seem to be affected. Not sure about other options.

### PR DESCRIPTION
**Pull request by ngosang**

### CODE REVIEW

**Comment by ngosang** [comment]
@ilike2burnthing I can't enable "enhanced logging" checkbox when we restart the server to apply the proxy configuration. In this PR I'm disabling the debug log to make it consistent with the UI. Take a look and tell me if this improves or not the current implementation.

File: src/Jackett.Server/Controllers/ServerConfigurationController.cs [hunk 1]
```diff
@@ -78,7 +78,8 @@ public IActionResult UpdateConfig([FromBody]Common.Models.DTO.ServerConfig confi
             var saveDir = config.blackholedir;
             var updateDisabled = config.updatedisabled;
             var preRelease = config.prerelease;
-            var logging = config.logging;
+            var enhancedLoging = config.logging;
```
Comment (ilike2burnthing) [inline src/Jackett.Server/Controllers/ServerConfigurationController.cs:81]:
```suggestion
            var enhancedLogging = config.logging;
```

File: src/Jackett.Server/Controllers/ServerConfigurationController.cs [hunk 2]
```diff
@@ -188,10 +186,17 @@ public IActionResult UpdateConfig([FromBody]Common.Models.DTO.ServerConfig confi
...
+                // configuration. when the server restarts the UI is inconsistent with the active log level
+                // https://github.com/Jackett/Jackett/issues/8315
+                setEnhancedLogLevel(false);
+
                 Thread.Sleep(500);
                 logger.Info("Restarting webhost due to configuration change");
                 Helper.RestartWebHost();
             }
+            else
+                setEnhancedLogLevel(enhancedLoging);
```
Comment (ilike2burnthing) [inline src/Jackett.Server/Controllers/ServerConfigurationController.cs:199]:
```suggestion
                setEnhancedLogLevel(enhancedLogging);
```

**Comment by ilike2burnthing** [comment]
Working in Windows, thanks for this

**Pull request merged**